### ISSUE TITLE
False domain error reported when name of expected verb instead refers to a noun

### ISSUE THREAD

**Issue by cchando** [jsoftware/jsource#207]
<pre>
   f=: N+]
   N=: 3
   f 1
|domain error in N, executing monad N
|       f 1
</pre>

Because <code>N</code> wasn't defined at the time that <code>f</code> was, it was assumed to be a verb. When the execution of [<code>N</code> as a verb], an error was raised because it instead referred to a noun. 

This is reported as a domain error, which it is confusing; it should be interpreted as a value error since it concerns N's part-of-speech only. It might say e.g. 
<pre>
|value error in N, executing monad N: N was expected to be a verb, but is defined as a noun
|       f 1
</pre>

Perhaps better would be to instead attempt to interpret <code>N</code> as a noun and apply the containing expression accordingly, upon the interpreter seeing that N is a noun (or just as a general fallback to attempted verb interpretations). This latter suggestion might not be feasible, depending on implementation details.

**Comment by HenryHRich** [comment]
Agreed that domain error is misleading. 'value error'means undefined name.
This is a /pun in type/, where JE analyzed a phrase using the parts of
speech of the defined names, and later, when the result of the analysis was
to be used, a name had a different type.  Worth a better message, I agree.

Henry Rich

**Comment by HenryHRich** [comment]
New message coming in next beta

Henry Rich
